tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/DefaultClassPathClassifier.java

```diff
@@ -219,6 +236,77 @@ public ArtifactUrlClassification classify(ClassPathClassifierContext context) {
   }
 
   /**
+   * Finds if there are {@value #SERVICE_PROPERTIES_FILE_NAME} in classpath in order to build a {@link ServiceUrlClassification}
+   * for each artifact that contains the {@value #SERVICE_PROPERTIES_FILE_NAME} file, {@cvalue #SERVICE_PROVIDER_CLASS_NAME} will
+   * be used as {@link ArtifactClassLoader#getArtifactName()}. Once an artifact is identified the dependencies are going to be
+   * collected from the graph using Maven {@code provided} scope.
+   *
+   * @param extendedContext {@link ExtendedClassPathClassifierContext} that holds the data needed for classifying the artifacts
+   * @return a {@link List} of {@link ServiceUrlClassification}s that would be the one used for the plugins class loaders.
+   */
+  protected List<ServiceUrlClassification> buildServicesUrlClassification(final ExtendedClassPathClassifierContext extendedContext) {
+    URLClassLoader classLoader = extendedContext.getClassificationContext().getClassPathClassLoader();
+    PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver(classLoader);
+    List<ServiceUrlClassification> serviceUrlClassifications = newArrayList();
+    Resource[] resources;
+    try {
+      resources = resolver.getResources(CLASSPATH_ALL_URL_PREFIX + "/META-INF/" + SERVICE_PROPERTIES_FILE_NAME);
+      logger.debug("Discovered " + resources.length + " services, processing them to build the URLs list for each one");
```

Comment:
Use a parameter for the log message, otherwise check if the debug level is enabled